test/unit/services/azuresqldb/provision-spec.js

```diff
@@ -190,144 +190,328 @@ describe('SqlDb - Provision - Execution - Database that does not previously exis
+            instance_id: 'e2778b98-0b6b-11e6-9db3-000d3a002ed5',
+            plan_id: "3819fdfa-0aaa-11e6-86f4-000d3a002ed5",
+            parameters: {      // developer's input parameters file
+                resourceGroup: 'sqldbResourceGroup',
+                sqlServerName: 'golive',
+                allowSqlServerFirewallRule: {
+                    ruleName: 'new rule',
+                    endIpAddress: '131.107.159.102'
+                },
+                sqlServerCreateIfNotExist: true,
+                sqlServerParameters: {
+                    location: 'westus',
+                    properties: {
+                        administratorLogin: 'greg',
+                        administratorLoginPassword: 'P@ssw0rd!'
+                    }
+                },
+                sqldbName: 'goliveSqlDb',
+                sqldbParameters: {
+                    properties: {
+                        collation: 'SQL_Latin1_General_CP1_CI_AS'
+                    }
+                }
+            },
+            azure: azure
+        };
+        cp = new cmdProvision(log, validParams);
+        cp.fixupParameters();
+    });
+
+    describe('Parameter validation should fail if ...', function () {
+        it('incorrect firewall rule specs are given - no start ip', function (done) {
+            (cp.allValidatorsSucceed()).should.equal(false);
+            done();
         });
     });
 });
 
+describe('SqlDb - Provision - Firewall rules', function () {
+    var validParams = {};
+    var cp;
+
+    before(function () {
+        validParams = {
+            instance_id: 'e2778b98-0b6b-11e6-9db3-000d3a002ed5',
+            plan_id: "3819fdfa-0aaa-11e6-86f4-000d3a002ed5",
+            parameters: {      // developer's input parameters file
+                resourceGroup: 'sqldbResourceGroup',
+                sqlServerName: 'golive',
+                allowSqlServerFirewallRule: {
+                    ruleName: 'new rule',
+                    startIpAddress: '131.107.159.102'
+                },
+                sqlServerCreateIfNotExist: true,
+                sqlServerParameters: {
+                    location: 'westus',
+                    properties: {
+                        administratorLogin: 'greg',
+                        administratorLoginPassword: 'P@ssw0rd!'
+                    }
+                },
+                sqldbName: 'goliveSqlDb',
+                sqldbParameters: {
+                    properties: {
+                        collation: 'SQL_Latin1_General_CP1_CI_AS'
+                    }
+                }
+            },
+            azure: azure
+        };
+        cp = new cmdProvision(log, validParams);
+        cp.fixupParameters();
+    });
+
+    describe('Parameter validation should succeed if ...', function () {
+        it('correct firewall rule specs are given - no end ip', function (done) {
+            (cp.allValidatorsSucceed()).should.equal(true);
+            done();
+        });
+    });
+});
```

Comment:
I think, these four cases can share one `describe('SqlDb - Provision - Firewall rules', function () {`.
For examxple:
```
describe('SqlDb - Provision - Firewall rules', function () {
  describe('Parameter validation should fail if ...', function () {
    before(function() {
      ...
    });
    it('incorrect firewall rule specs are given - no start ip', function (done) {
      (cp.allValidatorsSucceed()).should.equal(false);
      done();
    });
  });

  describe('Parameter validation should succeed if ...', function () {
    it('correct firewall rule specs are given - no end ip', function (done) {
      ....
    });
  });

  ...
});
```